update operator to use kops manifests

scoleman-ib opened the pull request:
Addressing [Issue 3](https://github.com/infobloxopen/cluster-operator/issues/3).

- Update CRD / API to use Kops manifest yaml
- Update state machine to use kops replace instead of kops create to utilize the Kops Manifest
- Update default KopsConfig to provide only Name and StateStore
- Update Cluster.yaml.in to provide a sample CR with the kops manifest for testing
- Update make deploy/Cluster.yaml to also fill in the ssh key with the key provided at ssh/kops.pub

The update also addressed [Issue 8](https://github.com/infobloxopen/cluster-operator/issues/8) as the pattern for both new and updated clusters should be kops replace -> kops update.  Noticed  updating the CR does not reset the phase from DONE, would be a TODO item to finish up that issue.

scoleman-ib committed:
update operator to use kops manifests

update operator to use kops manifests

seizadi reviewed (commented):
In kops.go you still have my FIXME
		// FIXME - Add in when we switch to kops config
Was this intentional, since you did add the lifecycle-overrides?

scoleman-ib commented:
No, I meant to remove the FIXME since I added the lines in.  Missed it when cleaning up, I'll remove the comment as well.

scoleman-ib commented on kops/kops.go:
```diff
@@ -133,8 +166,8 @@ func (k *KopsCmd) UpdateCluster(cluster clusteroperatorv1alpha1.KopsConfig) erro
 		" --name=" + cluster.Name +
 		// FIXME - Add in when we switch to kops config
 		// https://github.com/kubernetes/kops/blob/master/docs/iam_roles.md#use-existing-aws-instance-profiles
```
```suggestion
```

hryan-ib commented:
Shared an issue with Shaun about the tmp file not existing on a fresh clone and causing an error. Other than that, running the code looks good to me so far

scoleman-ib committed:
create tmp during write of kops manifest

scoleman-ib commented:
Updated to fix the issue, added a temporary function that both creates the tmp directory and writes the file to that directory.  Function is temporary as Haylee and I are exploring using STDIN instead of writing the manifest to file for [issue 16](https://github.com/infobloxopen/cluster-operator/issues/16).
